Utils/azuretools-core/src/com/microsoft/azuretools/authmanage/AdAuthManager.java

```diff
@@ -137,6 +139,21 @@ public synchronized boolean tryRestoreSignIn(@NotNull AuthMethodDetails authMeth
         return false;
     }
 
+    public AuthResult deviceLogin(final AuthenticationCallback<AuthenticationResult> callback) throws IOException {
```

Comment:
I suggest not to mix `deviceLogin` with `AdAuthManager`

Reply:
Explained above. Use `DCAuthManager` to handle the device login.